### ISSUE TITLE
Add pool_timeout as a parameter in HTTPConnectionPool

### ISSUE THREAD

**Issue by csala** [shazow/urllib3#1101]
This issue is a follow up of [this issue](https://github.com/kennethreitz/requests/issues/3825#issuecomment-274070742) opened in the [requests repository](https://github.com/kennethreitz/requests)

As indicated in the file [requests/packages/README.rst](https://github.com/kennethreitz/requests/blob/master/requests/packages/README.rst), if this is accepted, the changes will be first implemented and pushed here.

**Comment by Lukasa** [comment]
Is there a reason we need to do this? Requests can simply pass the appropriate option to urlopen or `new_conn`.

**Comment by csala** [comment]
@Lukasa You are right, strictly speaking there is no need to put it here and requests can pass it to urlopen.

But since this parameter controls the connection pool behaviour exactly as `block` and `maxsize` do, I think it's better to follow the same approach.

**Comment by Lukasa** [comment]
Yeah, I think that makes some sense.
